src/screens/SSICredentialsRequiredScreen/index.tsx

```diff
@@ -103,54 +112,75 @@ const SSICredentialsRequiredScreen: FC<Props> = (props: Props): JSX.Element => {
+          (branding: ICredentialBranding): boolean => branding.vcHash === uniqueVC.hash,
+        );
+        const credentialSummary: ICredentialSummary = await toCredentialSummary(uniqueVC, credentialBranding?.localeBranding);
+        const rawCredential: OriginalVerifiableCredential = await getOriginalVerifiableCredential(uniqueVC.verifiableCredential);
+        const isSelected: boolean = selectedCredentials
+          .get(inputDescriptorId)!
+          .some(
+            (matchedVC: UniqueVerifiableCredential) =>
+              matchedVC.verifiableCredential.id === uniqueVC.verifiableCredential.id ||
+              matchedVC.verifiableCredential.proof === uniqueVC.verifiableCredential.proof,
           );
-          const credentialSummary = await toCredentialSummary(uniqueVC, credentialBranding?.localeBranding);
-          const rawCredential = await getOriginalVerifiableCredential(uniqueVC.verifiableCredential);
-          const isSelected = selectedCredentials
-            .get(inputDescriptorId)!
-            .some(
-              matchedVC =>
-                matchedVC.verifiableCredential.id === uniqueVC.verifiableCredential.id ||
-                matchedVC.verifiableCredential.proof === uniqueVC.verifiableCredential.proof,
-            );
-          return {
-            hash: credentialSummary.hash,
-            id: credentialSummary.id,
-            credential: credentialSummary,
-            rawCredential: rawCredential,
-            isSelected: isSelected,
-          };
-        }),
-      ),
-      purpose: inputDescriptorPurpose,
-      // TODO move this to a function, would be nicer
-      onSelect: async (hashes: Array<string>) => {
-        const selectedVCs = availableCredentials.get(inputDescriptorId)!.filter(vc => hashes.includes(vc.hash));
-        selectedCredentials.set(inputDescriptorId, selectedVCs);
-        const newSelection = new Map<string, Array<UniqueVerifiableCredential>>();
-        for (const [key, value] of selectedCredentials) {
-          newSelection.set(key, value);
+        return {
+          hash: credentialSummary.hash,
+          id: credentialSummary.id,
+          credential: credentialSummary,
+          rawCredential: rawCredential,
+          isSelected: isSelected,
+        };
+      }),
+    );
+
+    const onSelectCredential = async (hashes: Array<string>): Promise<void> => {
+      const selectedVCs: Array<UniqueVerifiableCredential> = availableCredentials
+        .get(inputDescriptorId)!
+        .filter((vc: UniqueVerifiableCredential) => hashes.includes(vc.hash));
+      selectedCredentials.set(inputDescriptorId, selectedVCs);
+      const newSelection: Map<string, Array<UniqueVerifiableCredential>> = new Map<string, Array<UniqueVerifiableCredential>>();
+      for (const [key, value] of selectedCredentials) {
+        newSelection.set(key, value);
+      }
+
+      setSelectedCredentials(newSelection);
+
+      // TODO improve, can we not keep the originals in the state
```

Comment:
Can't you do
```
       const selectedVCs: Array<UniqueVerifiableCredential> = availableCredentials
        .get(inputDescriptorId)!
        .filter((vc: UniqueVerifiableCredential) => hashes.includes(vc.hash))
        .map((vc: UniqueVerifiableCredential) => structuredClone(vc));
````
Not sure if you have structuredClone() available

Reply:
removed the todo, was old anyways. besides that i did not want to spent to much time on refactoring logic